qiling/extensions/r2/r2.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+#
+# Cross Platform and Multi Architecture Advanced Binary Emulation Framework
+#
+
+import functools
+import json
+import ctypes
+import libr
+from dataclasses import dataclass, fields
+from enum import Enum
+from qiling.core import Qiling
+
+@dataclass(unsafe_hash=True)
+class Function:
+    name: str
+    offset: int
+    size: int
+    signature: str
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+@dataclass(unsafe_hash=True)
+class Section:
```

Comment:
Unless there is a fields post-processing or any additional functionality, these classes could be `typing.NamedTuple`.

Reply:
Yes. Now these classes have no special post-processing stuff. I use dataclass because it allows custom init function so I can pass all dict items got from r2 to construct the class without specifying fields.

Ref:
dataclass vs namedtuple: https://stackoverflow.com/questions/51671699/data-classes-vs-typing-namedtuple-primary-use-cases
create namedtuple only use a subset of arguments: https://stackoverflow.com/questions/17622419/creating-a-namedtuple-object-using-only-a-subset-of-arguments-passed